nightshift/datastore_transactions.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+from nightshift.datastore import Resource
+
+
+def insert_or_ignore(session, model, **kwargs):
+    """
+    Adds a new record to given table (model) or ignores if the same.
+
+    Args:
+        session:                sqlalchemy.Session instance
+        model:                  one of datastore table classes
+        kwargs:                 new record values as dictionary
+
+    Returns:
+        instance of inserted or duplicate record
+    """
+    instance = session.query(model).filter_by(**kwargs).first()
```

Comment:
Should these records always be unique? If so `one_or_none()` might be better here

Reply:
Looked up `one_or_none()` and yes, it will work better here.